tests/auth/test_client.py

```diff
@@ -0,0 +1,267 @@
+import openfga_sdk
+import pytest
+from openfga_sdk.api import open_fga_api
+from virtool_core.models.enums import Permission
+
+from virtool.auth.client import AuthorizationClient
+from virtool.auth.relationships import GroupPermission, UserPermission, GroupMembership
+from virtool.auth.utils import write_tuple, connect_openfga
+
+
+@pytest.fixture()
+def spawn_auth_client(mongo, data_layer, create_user):
+    async def func(
+        permissions=None,
+    ):
+        user_document = create_user(
+            user_id="test",
+            permissions=permissions,
+        )
+        await mongo.users.insert_one(user_document)
+
+        open_fga_client = await connect_openfga("localhost:8080", "http")
+
+        return AuthorizationClient(mongo, open_fga_client, data_layer)
+
+    return func
+
+
+@pytest.fixture()
+async def delete_store():
+    """
+    Delete the store.
+    """
+
+    configuration = openfga_sdk.Configuration(
+        api_scheme="http", api_host="localhost:8080"
+    )
+    api_client = open_fga_api.ApiClient(configuration)
+    api_instance = open_fga_api.OpenFgaApi(api_client)
+
+    response = await api_instance.list_stores()
+
+    for store in response.stores:
+        configuration.store_id = store.id
+        await api_instance.delete_store()
+
+
+@pytest.mark.parametrize("db", ["mongo", "open_fga"])
+@pytest.mark.parametrize("has_permission", [True, False])
+async def test_check(delete_store, db, has_permission, spawn_auth_client):
```

Comment:
Should be a two separate tests and maybe organized into a class: `TestCheck` with `test_mongo` and `test_openfga`.